pkg/gen/input/ami/ami.go

```diff
@@ -0,0 +1,67 @@
+package ami
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/devctl/pkg/gen/input"
+	"github.com/giantswarm/devctl/pkg/gen/input/ami/internal/file"
+	"github.com/giantswarm/devctl/pkg/gen/input/ami/internal/params"
+)
+
+type AMI struct {
+	config Config
+
+	booted bool
+	params params.Params
+}
+
+func New(config Config) (*AMI, error) {
+	err := config.Validate()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	f := &AMI{
+		config: config,
+	}
+
+	return f, nil
+}
+
+func (a *AMI) AMIFile() input.Input {
+	a.mustBooted()
+	return file.NewAMIInput(a.params)
+}
```

Comment:
I'm not very happy with `mustBooted` but I wanted the interface to be nice. When this is not booted then it's a bug.  It's not the error to handle or print to the end user IMO. Ideally we could do that in `New` but it would be against the principle we follow.